torch/csrc/autograd/python_function.cpp

```diff
@@ -50,22 +50,21 @@ PyObject* THPGradientEdgeClass = nullptr;
 #define THPFunction_assert(condition, ...) \
   if (!(condition)) {                      \
     THPUtils_setError(__VA_ARGS__);        \
-    throw python_error();                  \
+    throw_python_error();                  \
```

Comment:
I'm confused... Isn't your comment above saying that only the engine should do persist and the user code never has to do it?